src/main/java/com/google/devtools/build/lib/pkgcache/LoadingOptions.java

```diff
@@ -56,32 +66,32 @@
 
   @Option(
     name = "build_tag_filters",
-    converter = CommaSeparatedOptionListConverter.class,
+    converter = BooleanFormulaConverter.class,
     defaultValue = "",
     documentationCategory = OptionDocumentationCategory.UNCATEGORIZED,
     effectTags = {OptionEffectTag.UNKNOWN},
     help =
-        "Specifies a comma-separated list of tags. Each tag can be optionally "
-            + "preceded with '-' to specify excluded tags. Only those targets will be built that "
-            + "contain at least one included tag and do not contain any excluded tags. This option "
-            + "does not affect the set of tests executed with the 'test' command; those are be "
-            + "governed by the test filtering options, for example '--test_tag_filters'"
+        "A Boolean formula over rule tags that selects the targets to build. "
+            + "Example: '(server or client) and not experimental'. "
+            + "See --test_tag_filters for details."
   )
-  public List<String> buildTagFilterList;
+  public Expression buildTagFilter;
 
   @Option(
     name = "test_tag_filters",
-    converter = CommaSeparatedOptionListConverter.class,
+    converter = BooleanFormulaConverter.class,
     defaultValue = "",
     documentationCategory = OptionDocumentationCategory.UNCATEGORIZED,
     effectTags = {OptionEffectTag.UNKNOWN},
     help =
-        "Specifies a comma-separated list of test tags. Each tag can be optionally "
-            + "preceded with '-' to specify excluded tags. Only those test targets will be "
-            + "found that contain at least one included tag and do not contain any excluded "
-            + "tags. This option affects --build_tests_only behavior and the test command."
+        "A Boolean formula over test tags that selects the tests to run. "
+            + "Example: '(release or smoke) and not flaky'. Formulas have the form: "
+            + "expr = IDENT | expr 'or' expr | expr 'and' expr | 'not' expr | '(' expr ')'. "
+            + "IDENT is defined as a Starlark identifier, with the same reserved words. "
+            + "The legacy syntax, a comma-separated list of tags, implicitly joined as if by 'or', "
+            + "each optionally preceded by '-' for negation, will be removed after Jan 1, 2021."
```

Comment:
> ... The legacy syntax...  will be removed after Jan 1, 2021.

Sorry, I saw this PR in passing and haven't been following the discussion or motivation. Was this removal discussed and agreed upon? If so, can you please link me? Also, is there a tracking Github issue for the removal? I ask because some things internally at Google use the legacy syntax and they will need to be updated. Thanks!

Reply:
@alandonovan suggested to drop the legacy syntax in a review comment. I don't think it has been discussed anywhere else. Personally, I don't have any strong opinion. For me it would be fine to keep the old syntax as well.

I suppose that the process for incompatible changes needs to be followed here? Will you write the tracking Github issue, or should I do it?

Reply:
Thanks.

I can file the tracking issue once this PR lands.